test/serialize_test.cpp

```diff
@@ -125,6 +130,21 @@ TEST_CASE(serialize_empty_struct)
     EXPECT(v.at("a").to<int>() == 1);
 }
 
+TEST_CASE(serialize_empty_optional)
```

Comment:
Add unittest for tuple as well

Reply:
The tuple is already tested in the `serialize_reflectable_type` test since it was added to the `reflectable_type` class.